django/core/checks/model_checks.py

```diff
@@ -29,6 +33,12 @@ def check_all_models(app_configs=None, **kwargs):
             )
         else:
             errors.extend(model.check(**kwargs))
+        for model_index in model._meta.indexes:
+            index_models.append(model)
+            index_names.append(model_index.name)
+        for model_constraint in model._meta.constraints:
+            constraint_models.append(model)
+            constraint_names.append(model_constraint.name)
```

Comment:
I would use the same mechanism as for the `E020` and models' labels instead of `__name__`'s, i.e.
```python
indexes = defaultdict(list)
constraints = defaultdict(list)
...
for model_index in model._meta.indexes:
    indexes[model_index.name].append(model._meta.label)
for model_constraint in model._meta.constraints:
    constraints[model_constraint.name].append(model._meta.label)
```